pyomo/solvers/plugins/solvers/cplex_direct.py

```diff
@@ -287,7 +382,49 @@ def _set_instance(self, model, kwds={}):
                             "by overwriting its bounds in the CPLEX instance."
                             % (var.name, self._pyomo_model.name,))
 
-    def _add_constraint(self, con):
+    def _add_block(self, block):
+        with _VariableData(self._solver_model) as cplex_var_data:
+            for var in block.component_data_objects(
+                ctype=pyomo.core.base.var.Var, descend_into=True, active=True, sort=True
```

Comment:
Here (and elsewhere), I believe that you can make this more performant (avoid unnecessary sorting) by using `sort=SortComponents.deterministic` instead of `True`.

Reply:
Happy to change this, just note it would be a behaviour change whereas the rest of this MR maintains identical behaviour to the existing implementation.

Reply:
Good point.  Preserving behavior is a good thing.  We can revisit sorting later (probably as part of #677 / #1030).